src/components/metric-chart.component.ts

```diff
@@ -16,37 +20,38 @@ import { ScatterLineChart } from '../model/scatterLine'
 import { HistogramChart } from '../model/histogram'
 import { RhqBarChart } from '../model/rhq-bar'
 import { MultiLineChart } from '../model/multi-line'
-import { determineXAxisTicksFromScreenWidth, determineYAxisTicksFromScreenHeight,
-   xAxisTimeFormats, determineYAxisGridLineTicksFromScreenHeight } from '../util/utility'
+import {
+  determineXAxisTicksFromScreenWidth, determineYAxisTicksFromScreenHeight,
+  xAxisTimeFormats, determineYAxisGridLineTicksFromScreenHeight
+} from '../util/utility'
 import { createAlertBoundsArea, createAlertLine } from '../util/alerts'
 import { createDataPoints } from '../util/features'
 import { showForecastData } from '../util/forecast'
 
-declare let d3: any;
 declare let moment: any;
+declare let d3: any;
 declare let console: any;
 
-const debug = true;
+const debug = false;
 
 const DEFAULT_Y_SCALE = 10;
 const X_AXIS_HEIGHT = 25; // with room for label
-const HOVER_DATE_TIME_FORMAT = 'MM/DD/YYYY h:mm a';
-const MARGIN = { top: 10, right: 5, bottom: 5, left: 90 }; // left margin room for label
+const HOVER_DATE_TIME_FORMAT = 'MM/DD/YYYY h:mm:ss a';
+const MARGIN = { top: 0, right: 0, bottom: 0, left: 0 }; // left margin room for label
 
 @Component({
   selector: 'hk-metric-chart',
   template: `<div #target class='hawkular-charts'></div>`
 })
-export class MetricChartComponent implements OnChanges {
+export class MetricChartComponent implements OnInit, OnDestroy, OnChanges {
```

Comment:
This component is way too huge. I guess refactoring it using couple of services might be better.